src/Integration.Vsix/ErrorList/SonarErrorListDataSource.cs

```diff
@@ -212,10 +212,18 @@ public void Refresh(IEnumerable<string> affectedFilePaths)
                 foreach (var factory in factories)
                 {
                     var snapshot = factory.CurrentSnapshot;
-                    
-                    if (snapshot.FilesInSnapshot.Any(snapshotPath => affectedFilePaths.Any(affected => PathHelper.IsMatchingPath(snapshotPath, affected))))
+                    var isSnapshotAffected = snapshot.FilesInSnapshot.Any(snapshotPath => affectedFilePaths.Any(affected => PathHelper.IsMatchingPath(snapshotPath, affected)));
+
+                    if (isSnapshotAffected)
                     {
-                        snapshot.IncrementVersion();
+                        if (snapshot.HasNonNavigableIssues)
+                        {
+                            factory.UpdateSnapshot(factory.CurrentSnapshot.CreateUpdatedSnapshot());
```

Comment:
After extensive testing, I believe this is the "best" we could do with.
I tested 2 other approaches:
1. Removing the non navigable issues when IncrementVersion is called.
2. Managing 2 lists in the snapshot, Issues (for a history of all issues) and VisibleIssues (for GetValue, Count, etc).

Both approaches failed in IndexOf, returning garbage results in various scenarios, completely throwing off secondary locations. It was way worse in some cases than master, i.e. selecting a navigable issue and invalidating a non navigable issue would throw the selection, even though the user selected an issue that was not impacted.
You can read the reasoning behind this here: https://github.com/SonarSource/sonarlint-visualstudio/blob/67dd4681e3ed08911b074de1883379e8371f3c9a/src/Integration.Vsix/ErrorList/IssuesSnapshot.cs#L287

So in the end I opted for creating a new snapshot. The reasoning is that the issues _are_ changed: it's not the same exact set of issues anymore, so it makes sense to create a new one. This allows IndexOf to behave properly, as I can reliably find the old issue in the new snapshot.